pkg/kubectl-argo-rollouts/cmd/terminate/terminate.go

```diff
@@ -0,0 +1,97 @@
+package terminate
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	types "k8s.io/apimachinery/pkg/types"
+
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/options"
+)
+
+const (
+	terminatePatch = `{"spec":{"terminate":true}}`
+)
+
+// NewCmdTerminate returns a new instance of an `argo rollouts terminate` command
+func NewCmdTerminate(o *options.ArgoRolloutsOptions) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "terminate <analysisRun|experiment> RESOURCE",
+		Short: "Terminate a analysisRun or experiment",
+		Example: o.Example(`
+  # Terminate an analysisRun
+  %[1]s terminate analysisRun ANALYSISRUN
+  # Terminate a failed experiment
+  %[1]s terminate experiment EXPERIMENT
+`),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			return o.UsageErr(c)
+		},
+	}
+	cmd.AddCommand(NewCmdTerminateAnalysisRun(o))
+	cmd.AddCommand(NewCmdTerminateExperiment(o))
+	return cmd
+}
+
+// NewCmdTerminateAnalysisRun returns a new instance of an `argo rollouts terminate analysisRun` command
+func NewCmdTerminateAnalysisRun(o *options.ArgoRolloutsOptions) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:     "analysisRun ANALYSISRUN",
```

Comment:
analysisRun should be all lowercase since this is consistent on how it will be used in kubectl (e.g. `kubectl get analysisruns`), and also because CLIs generally do not use camel case.